apps/ary/views.py

```diff
@@ -166,28 +168,19 @@ def get(self, request, version=None):
 
         options = {}
 
-        def _filter_by_projects(qs, projects):
-            for p in projects:
-                qs = qs.filter(project=p)
-            return qs
-
-        if (fields is None or 'created_by' in fields):
-            created_by = _filter_by_projects(User.objects, projects)
-            options['created_by'] = [
-                {
-                    'key': user.id,
-                    'value': user.profile.get_display_name(),
-                } for user in created_by.distinct()
-            ]
-
-        if (fields is None or 'project' in fields):
-            projects = Project.get_for_member(request.user)
-            options['project'] = [
-                {
-                    'key': project.id,
-                    'value': project.title,
-                } for project in projects.distinct()
-            ]
+        options['created_by'] = Assessment.objects.filter(
+            project__in=projects
+        ).distinct().values('created_by').annotate(
+            key=models.F('created_by'),
+            value=models.F('created_by__username')
+        ).values('key', 'value')
+
+        options['project'] = [
+            {
+                'key': project.id,
+                'value': project.title,
+            } for project in projects.distinct()
+        ] if (fields is None or 'project' in fields) else []
```

Comment:
Let's do this way (using get_display_name). Will do query optimization in graphql.
```suggestion
        if (fields is None or 'created_by' in fields):
            assessment_qs = Assessment.objects.filter(project__in=projects)
            options['created_by'] = [
                {
                    'key': user.id,
                    'value': user.profile.get_display_name(),
                }
                for user in User.objects.filter(pk__in=assessment_qs.distinct().values('created_by')).select_related('profile')
            ]

        if (fields is None or 'project' in fields):
            options['project'] = [
                {
                    'key': project.id,
                    'value': project.title,
                } for project in projects.distinct()
            ]
```